packages/flutter/test/widgets/focus_manager_test.dart

```diff
@@ -510,27 +510,250 @@ void main() {
+      scope1Attachment.reparent(parent: tester.binding.focusManager.rootScope);
+      scope2Attachment.reparent(parent: tester.binding.focusManager.rootScope);
+      parent1Attachment.reparent(parent: scope1);
+      parent2Attachment.reparent(parent: scope2);
+      child1Attachment.reparent(parent: parent1);
+      child2Attachment.reparent(parent: parent1);
+      child3Attachment.reparent(parent: parent2);
+      child4Attachment.reparent(parent: parent2);
 
+      // Build up a history.
+      child4.requestFocus();
+      await tester.pump();
+      child2.requestFocus();
+      await tester.pump();
+      child3.requestFocus();
+      await tester.pump();
       child1.requestFocus();
       await tester.pump();
       expect(scope1.focusedChild, equals(child1));
-      expect(parent2.children.contains(child1), isFalse);
+      expect(scope2.focusedChild, equals(child3));
+
+      child1.unfocus(disposition: UnfocusDisposition.scope);
+      await tester.pump();
+      // Focused child doesn't change.
+      expect(scope1.focusedChild, isNull);
+      expect(scope2.focusedChild, equals(child3));
+      // Focus does change.
+      expect(scope1.hasPrimaryFocus, isTrue);
+      expect(scope2.hasFocus, isFalse);
+      expect(child1.hasPrimaryFocus, isFalse);
+      expect(child2.hasPrimaryFocus, isFalse);
 
-      scope1.unfocus();
+      // Can re-focus child.
+      child1.requestFocus();
+      await tester.pump();
+      expect(scope1.focusedChild, equals(child1));
+      expect(scope2.focusedChild, equals(child3));
+      expect(scope1.hasFocus, isTrue);
+      expect(scope2.hasFocus, isFalse);
+      expect(child1.hasPrimaryFocus, isTrue);
+      expect(child3.hasPrimaryFocus, isFalse);
+
+      // The same thing happens when unfocusing a second time.
+      child1.unfocus(disposition: UnfocusDisposition.scope);
       await tester.pump();
       expect(scope1.focusedChild, isNull);
+      expect(scope2.focusedChild, equals(child3));
+      expect(scope1.hasPrimaryFocus, isTrue);
+      expect(scope2.hasFocus, isFalse);
       expect(child1.hasPrimaryFocus, isFalse);
+      expect(child2.hasPrimaryFocus, isFalse);
+
+      // When the scope gets unfocused, then its parent scope (the root scope)
+      // gets focus, but it doesn't mess with the focused children.
+      child1.requestFocus();
+      await tester.pump();
+      scope1.unfocus(disposition: UnfocusDisposition.scope);
+      await tester.pump();
+      expect(scope1.focusedChild, equals(child1));
+      expect(scope2.focusedChild, equals(child3));
       expect(scope1.hasFocus, isFalse);
+      expect(scope2.hasFocus, isFalse);
+      expect(child1.hasPrimaryFocus, isFalse);
+      expect(child3.hasPrimaryFocus, isFalse);
+      expect(FocusManager.instance.rootScope.hasPrimaryFocus, isTrue);
+    });
+    testWidgets('Unfocus with disposition scope works properly when some nodes are unfocusable', (WidgetTester tester) async {
```

Comment:
can you include a similar test for UnfocusDisposition.previouslyFocusedChild

Reply:
Oh, yes, I actually thought I already did that.